Add documentation for Transmission component (#7816)
home-assistant/home-assistant.github.io: source/_components/transmission.markdown

---
layout: page
title: "Transmission"
description: "Instructions on how to integrate Transmission within Home Assistant."
date: 2018-12-12 22:44
sidebar: true
comments: false
sharing: true
footer: true
logo: transmission.png

> ha_category: Downloading

Reviewer: ```suggestion
ha_category: 
  - Downloading
  - Switch
  - Sensor
```

> ha_release: 0.85

Reviewer 2: 0.87

ha_iot_class: "Local Polling"

> ---

Reviewer: ```suggestion
redirect_from:
  - /components/switch.transmission/
  - /components/sensor.transmission/
```

</p>

If everything is set up correctly, the details will show up in the frontend.

<p class='img'>
  <img src='{{site_root}}/images/components/transmission/transmission.png' />
</p>

## {% linkable_title Configuration %}

To enable this sensor, add the following lines to your `configuration.yaml`:

```yaml
transmission:
  host: 192.168.1.1
```

{% configuration %}
host:
  description: This is the IP address of your Transmission daemon, e.g., `192.168.1.1`.
  required: true
  type: string
port:
  description: The port your Transmission daemon uses.
  required: false
  type: integer
  default: 9091
name:
  description: The name to use when displaying this Transmission instance in the frontend.
  required: false
  type: string
username:
  description: Your Transmission username, if you use authentication.
  required: false
  type: string
password:
  description: Your Transmission password, if you use authentication.
  required: false
  type: string
turtle_mode:
  description: If enabled, it creates a switch entity to control the 'Alternative Speed Limits' (aka 'Turtle mode') setting.
  required: false
  type: boolean

> monitored_variables:

Reviewer 3: This is now `monitored_conditions`

> monitored_conditions:

Reviewer: ```suggestion
scan_interval:
  description: How frequently to query for new data. Defaults to 120 seconds.
  required: false
  type: integer
```

Above the `Monitored Conditions`, see also: https://github.com/home-assistant/home-assistant.io/pull/8344

>   description: Conditions to display in the frontend. Every one of them will act as a sensor.

Reviewer 2: This doesn't have anything to do with the frontend per se. This sets what conditions will be tracked as entitites.

  required: false
  type: map
  keys:
    current_status:
      description: The status of your Transmission daemon.
    download_speed:
      description: The current download speed [MB/s].
    upload_speed:
      description: The current upload speed [MB/s].
    active_torrents:
      description: The current number of active torrents.
    paused_torrents:
      description: The current number of paused torrents.
    total_torrents:
      description: The total number of torrents present in the client.
    started_torrents:
      description: The current number of started torrents (downloading).
    completed_torrents:
      description: The current number of completed torrents (seeding)
{% endconfiguration %}

## {% linkable_title Event Automation %}

> The Transmission component is continuously monitoring the status of torrents in the target client. Once a torrent is started or completed, an event is triggered on the Home Assistant Bus, allows to implement any kind of automation.

Reviewer 2: an event is triggered on the Home Assistant event bus, which allows to implement any kind of automation.

---
layout: page
title: "Transmission"
description: "Instructions on how to integrate Transmission within Home Assistant."
date: 2018-12-12 22:44
sidebar: true
comments: false
sharing: true
footer: true
logo: transmission.png
ha_category: Downloading
ha_release: 0.85
ha_iot_class: "Local Polling"
---

The `transmission` component allows you to monitor your downloads with [Transmission](http://www.transmissionbt.com/) from within Home Assistant and setup automation based on the information.

## {% linkable_title Setup %}

To use the monitoring, your transmission client needs to allow remote access. If you are running the graphical transmission client (transmission-gtk) go to **Edit** -> **Preferences** and choose the tab **Remote**. Check **Allow remote access**, enter your username and your password, and uncheck the network restriction as needed.

<p class='img'>
  <img src='{{site_root}}/images/components/transmission/transmission_perf.png' />
</p>

> If everything is setup correctly, the details will show up in the frontend.

Reviewer 4: ```suggestion
If everything is set up correctly, the details will show up in the frontend.
```

<p class='img'>
  <img src='{{site_root}}/images/components/transmission/transmission.png' />
</p>

## {% linkable_title Configuration %}

To enable this sensor, add the following lines to your `configuration.yaml`:

```yaml
transmission:
  host: 192.168.1.1

>   username: user

Reviewer 4: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

>   password: psw

Reviewer 4: Same as above.

>   port: 8181

Reviewer 4: Same as above.

>   monitored_variables:

Reviewer 4: Same as above.

  - current_status
  - completed_torrents
  - started_torrents
```

{% configuration %}
host:
  description: This is the IP address of your Transmission daemon, e.g., 192.168.1.1.
  required: true
  type: string
port:
  description: The port your Transmission daemon uses.
  required: false
  type: integer
  default: 9091
name:
  description: The name to use when displaying this Transmission instance in the frontend.
  required: false
  type: string
username:
  description: Your Transmission username, if you use authentication.
  required: false
  type: string
password:
  description: Your Transmission password, if you use authentication.
  required: false
  type: string
monitored_variables:

>   description: Conditions to display in the frontend. Everyone of them will act as sensor.

Reviewer 4: ```suggestion
  description: Conditions to display in the frontend. Every one of them will act as a sensor.
```

  description: If enabled, it creates a switch entity to control the 'Alternative Speed Limits' (aka 'Turtle mode') setting.
  required: false
  type: boolean

> scan_interval:

Reviewer 2: This isn't an option at the moment. There's an open PR to add it, but that has its own docs PR.

Author: What should I do?

Reviewer 2: Remove the `scan_interval` option here.
